src/js-form-validator.ts

```diff
@@ -48,20 +49,27 @@ export class FormValidator {
       if (inputs.length <= 0) return false;
 
       Array.prototype.forEach.call(inputs, (i: HTMLInputElement) => {
-        if (this.validate_rule('checkbox', i.checked))
+        if (this.validate_rule('checkbox', i.checked, allowEmpty))
           numChecked++;
       });
 
       if (numChecked > 0)
         valid = 1;
     }
+    else if (type === 'select') {
+      const selectedOption = selectElement.options[selectElement.selectedIndex];
+      if (selectedOption.getAttribute('disabled') !== null)
```

Comment:
You could simplify this to `valid = (selectedOption.getAttribute('disabled') !== null) ? -1 : 1;`